### PR TITLE
Host preflight warnings

### PR DESCRIPTION
**Pull request by emosbaugh**

### CODE REVIEW

File: pkg/preflight/assets/host-preflights.yaml [hunk 1]
```diff
@@ -20,17 +20,23 @@ spec:
...
         checkName: "Number of CPUs"
         outcomes:
-          - fail:
+          - warn:
               when: "count < 4"
-              message: At least 4 CPU cores are required
+              message: At least 4 CPU cores are recommended
+          - fail:
+              when: "count < 2"
+              message: At least 2 CPU cores are required
```
Comment (markpundsack) [inline pkg/preflight/assets/host-preflights.yaml:28]:
```suggestion
              message: At least 2 CPU cores are required, and 4 CPU cores are recommended
```

File: pkg/preflight/assets/host-preflights.yaml [hunk 2]
```diff
@@ -20,17 +20,23 @@ spec:
...
         checkName: "Amount of Memory"
         outcomes:
-          - fail:
+          - warn:
               when: "< 8Gi"
-              message: At least 8Gi of memory is required
+              message: At least 8Gi of memory is recommended
+          - fail:
+              when: "< 4Gi"
+              message: At least 4Gi of memory is required
```
Comment (markpundsack) [inline pkg/preflight/assets/host-preflights.yaml:39]:
```suggestion
              message: At least 4Gi of memory is required, and 8Gi of memory is recommended
```

File: pkg/preflight/assets/host-preflights.yaml (hunk 1 above)
Comment (markpundsack) [reply pkg/preflight/assets/host-preflights.yaml:28]: It's a poor experience to be told a minimum, then reconfigure the machine to meet the minimum, only to be then told a higher recommended number, so just give them both at the same time.

My only question is, will it be clear to the user that they're below both? Does it say how many cores we detect?

File: pkg/preflight/builtin_test.go [hunk 3]
```diff
@@ -48,17 +48,23 @@ spec:
...
         checkName: "Number of CPUs"
         outcomes:
-          - fail:
+          - warn:
               when: "count < 4"
-              message: At least 4 CPU cores are required
+              message: At least 4 CPU cores are recommended
+          - fail:
+              when: "count < 2"
+              message: At least 2 CPU cores are required
```
Comment (markpundsack) [inline pkg/preflight/builtin_test.go:56]:
```suggestion
              message: At least 2 CPU cores are required, and 4 CPU cores are recommended
```

File: pkg/preflight/builtin_test.go [hunk 4]
```diff
@@ -48,17 +48,23 @@ spec:
...
         checkName: "Amount of Memory"
         outcomes:
-          - fail:
+          - warn:
               when: "< 8Gi"
-              message: At least 8Gi of memory is required
+              message: At least 8Gi of memory is recommended
+          - fail:
+              when: "< 4Gi"
+              message: At least 4Gi of memory is required
```
Comment (markpundsack) [inline pkg/preflight/builtin_test.go:67]:
```suggestion
              message: At least 4Gi of memory is required, and 8Gi of memory is recommended
```

Comment (markpundsack) [reply pkg/preflight/builtin_test.go:67]: Could also do with with parens: `(8Gi memory recommended)`

**Review by areed — APPROVED** [review]

**Pull request merged**